iOS SDK README suggestions (#301)
klaviyo/klaviyo-swift-sdk: README.md

        pod 'KlaviyoSwift'
      end

>       target 'YourAppTarget' do

Reviewer: 👍

Klaviyo SDK will automatically handle the badge count associated with Klaviyo pushes. If you need to manually update the badge count to account for other notification sources, use the `KlaviyoSDK().setBadgeCount(:)` method, which will update the badge count and keep it in sync with the Klaviyo SDK. This method should be used instead of (rather than in addition to) setting the badge count using `UNUserNotificationCenter` and/or `UIApplication` methods.

## In-App Forms

> 

Reviewer 2: small nit since we use this tag on all other sections for version support
```suggestion
> ℹ️ In-app forms support is available in SDK version [4.2.0](https://github.com/klaviyo/klaviyo-swift-sdk/releases/tag/4.2.0) and higher
```

and then in the prerequisites section we could just have  
```
* Imported `KlaviyoSwift` and `KlaviyoForms` SDK modules and adding it to the app target (SDK version 4.2.0 and higher)
```

In-app forms are messages displayed to mobile app users while they are actively using an app. You can create new in-app forms in a drag-and-drop editor in the Sign-Up Forms tab in Klaviyo.

### Prerequisites

* Using Klaviyo SDK version [4.2.0](https://github.com/klaviyo/klaviyo-swift-sdk/releases/tag/4.2.0) and higher
* Imported `KlaviyoSwift` and `KlaviyoForms` SDK modules and adding it to the app target.

### Setup

Once you've created an in-app form in your Klaviyo account, to display it add the following code to your application.

```swift
    import KlaviyoSwift
    import KlaviyoForms
    ...

    // Look at the behaviour section below on where to add this code to.
    KlaviyoSDK()
        .initialize(with: "YOUR_KLAVIYO_PUBLIC_API_KEY")
        .registerForInAppForms()

>     // if registering else where after `KlaviyoSDK` is initlized

Reviewer: ```suggestion
    // if registering else where after `KlaviyoSDK` is initialized
```

    KlaviyoSDK().registerForInAppForms()
```

> ### Behaviour

Reviewer 2: ```suggestion
### Behavior
```

Author: oh come on

> Once `registerForInAppForms()` is called, the SDK will load form data for your account and display no more than one form within 10 seconds, based on the form frequency and trigger settings configured in your Klaviyo account.

Reviewer: ```suggestion
Once `registerForInAppForms()` is called, the SDK will load form data for your account and display no more than one form within 10 seconds, based on based on form targeting and behavior settings.```

You can call `registerForInAppForms()` any time after initializing with your company ID to control when and where in your app's UI a form can appear. It is safe to register multiple times per application session. The SDK will internally prevent multiple forms appearing at once.

Consider how often you want to register for forms. Below are some ideas on when forms can potentially be shown,

| **App State**                | **Lifecycle Method**                              |
|------------------------------|--------------------------------------------------|
| **App Launched (Cold Start)** | `application(_:didFinishLaunchingWithOptions:)` |
| **App Became Active**         | `applicationDidBecomeActive(_:)`                 |
| **Any App View Controller**         | `viewDidLoad()` |

For example, registering from a view controller is advisable as it increases the chance of your user seeing the form. However, be advised that this will be shown as soon as the form is ready in the SDK. Future versions of this product will provide more control in this regard.

> 

Reviewer: ```suggestion
**Note:** At this time, when device orientation changes any currently visible form is closed and will not be re-displayed automatically.
```

I don't love drawing attention to a deficiency, but I think its probably best to call it out.

Reviewer: we could bury it in the troubleshooting doc...

Reviewer: I'm not strongly attached to it, but I copied it forward from Dan's draft thinking that they might take it as a nudge to register again after a rotation.

Reviewer 3: I'm still on team keep it, I think this is a famous Android scenario that most developers are thinking about.

| **App Launched (Cold Start)** | `application(_:didFinishLaunchingWithOptions:)` | [See here](https://github.com/klaviyo/klaviyo-swift-sdk/blob/master/Examples/KlaviyoSwiftExamples/Shared/AppDelegate.swift#L41) |
| **App Became Active**         | `applicationDidBecomeActive(_:)`                 | [See here](https://github.com/klaviyo/klaviyo-swift-sdk/blob/master/Examples/KlaviyoSwiftExamples/Shared/AppDelegate.swift#L59) |
| **Any App View Controller**         | `viewDidLoad()` | [See here](https://github.com/klaviyo/klaviyo-swift-sdk/blob/master/Examples/KlaviyoSwiftExamples/Shared/MenuPageViewController.swift#L35) |

Registering from `applicationDidBecomeActive(_:)` is advisable as it increases the chance of your user seeing the form. However, be advised that this will be shown as soon as the form is ready in the SDK, so you may still need to condition this based on the user's context within your application. Future versions of this product will provide more control in this regard.

**Note:** At this time, when device orientation changes any currently visible form is closed and will not be re-displayed automatically.

### Deep linking

> Deep linking to a particular screen based on user action from an IAF is similar to handling deep links originating from push notifications. [Step 3](#step-3-implement-handling-deep-links-in-your-app) of the deep linking section outlines exactly how this can be achieved. Currently, if a web URL is included as the deep link, the SDK will open the URL in the default browser.

Reviewer: > Currently, if a web URL is included as the deep link, the SDK will open the URL in the default browser

I don't think this is true for universal links, if you want to get into that level of detail anyway.

Reviewer: By which I mean if the app is authorized to open universal links for a given web domain, then that should still open their app.

Reviewer: Perhaps
> For further information on how the deep link is handled, see [apple documentation](https://developer.apple.com/documentation/uikit/uiapplication/open(_:options:completionhandler:)).

        .registerForInAppForms()

    // if registering else where after `KlaviyoSDK` is initlized
    KlaviyoSDK().registerForInAppForms()
```

### Behaviour

Once `registerForInAppForms()` is called, the SDK will load form data for your account and display no more than one form within 10 seconds, based on the form frequency and trigger settings configured in your Klaviyo account.

You can call `registerForInAppForms()` any time after initializing with your company ID to control when and where in your app's UI a form can appear. It is safe to register multiple times per application session. The SDK will internally prevent multiple forms appearing at once.

Consider how often you want to register for forms. Below are some ideas on when forms can potentially be shown,

| **App State**                | **Lifecycle Method**                              |
|------------------------------|--------------------------------------------------|
| **App Launched (Cold Start)** | `application(_:didFinishLaunchingWithOptions:)` |
| **App Became Active**         | `applicationDidBecomeActive(_:)`                 |
| **Any App View Controller**         | `viewDidLoad()` |

> For example, registering from a view controller is advisable as it increases the chance of your user seeing the form. However, be advised that this will be shown as soon as the form is ready in the SDK. Future versions of this product will provide more control in this regard.

Author: @evan-masseau lets discuss this rec tomorrow. Copied what you had on android and updated to iOS but want to find a balance between keeping it simple vs. making a more detailed recommendation. Like I think on iOS may be `applicationDidBecomeActive(_:)`  is a good idea but of course it doesn't let you control not showing up on screens you  don't want it to on the other hand throwing it in a view controller/activity may lessen chances of someone actually seeing it and engaging with it.

Reviewer: >Future versions of this product will provide more control in this regard.

Good idea to nod at future improvements.

Reviewer: how about:

> However, be advised that this will be shown as soon as the form is ready in the SDK, so you may still need to condition this based on the user's context within your application. Future versions of this product will provide more control in this regard. 

adding this to android for now

Author: thanks, updated.

Klaviyo SDK will automatically handle the badge count associated with Klaviyo pushes. If you need to manually update the badge count to account for other notification sources, use the `KlaviyoSDK().setBadgeCount(:)` method, which will update the badge count and keep it in sync with the Klaviyo SDK. This method should be used instead of (rather than in addition to) setting the badge count using `UNUserNotificationCenter` and/or `UIApplication` methods.

## In-App Forms
> ℹ️ In-app forms support is available in SDK version [4.2.0](https://github.com/klaviyo/klaviyo-swift-sdk/releases/tag/4.2.0) and higher

[In-app forms](https://help.klaviyo.com/hc/en-us/articles/34567685177883) are messages displayed to mobile app users while they are actively using an app. You can create new in-app forms in a drag-and-drop editor in the Sign-Up Forms tab in Klaviyo.  Follow the instructions in this section to integrate forms with your app. The SDK will

> display forms according to targeting and behavior settings and collect delivery and engagement analytics automatically.

Reviewer 4: Slight tweak, but I feel like this reads a little better - "according to **their** targeting and behavior settings"

Reviewer: ```suggestion
display forms according to their targeting and behavior settings and collect delivery and engagement analytics automatically.
```

> display forms according to targeting and behavior settings and collect delivery and engagement analytics automatically.

Reviewer: ```suggestion
display forms according to their targeting and behavior settings and collect delivery and engagement analytics automatically.
```

### Prerequisites

* Using Klaviyo SDK version 4.2.0 and higher
* Imported `KlaviyoSwift` and `KlaviyoForms` SDK modules and adding it to the app target.

### Setup

> Once you've created an in-app form in your Klaviyo account, to display it add the following code to your application.

Reviewer 4: Can we simplify and just say "To display in-app forms add the following code to your application."

Reviewer: ```suggestion
To display in-app forms, add the following code to your application.
```

```swift
    import KlaviyoSwift
    import KlaviyoForms
    ...

    // Refer the behavior section for where to place this code.
    KlaviyoSDK()
        .initialize(with: "YOUR_KLAVIYO_PUBLIC_API_KEY")
        .registerForInAppForms()

    // if registering else where after `KlaviyoSDK` is initialized
    KlaviyoSDK().registerForInAppForms()
```

### Behavior

> Once `registerForInAppForms()` is called, the SDK will load form data for your account and display no more than one form within 10 seconds,  based on form targeting and behavior settings.

Reviewer: ```suggestion
Once `registerForInAppForms()` is called, the SDK will load form data for your account and display no more than one form within 15 seconds,  based on form targeting and behavior settings.
```

| **App Launched (Cold Start)** | `application(_:didFinishLaunchingWithOptions:)` | [See here](https://github.com/klaviyo/klaviyo-swift-sdk/blob/master/Examples/KlaviyoSwiftExamples/Shared/AppDelegate.swift#L41) |
| **App Became Active**         | `applicationDidBecomeActive(_:)`                 | [See here](https://github.com/klaviyo/klaviyo-swift-sdk/blob/master/Examples/KlaviyoSwiftExamples/Shared/AppDelegate.swift#L59) |
| **Any App View Controller**         | `viewDidLoad()` | [See here](https://github.com/klaviyo/klaviyo-swift-sdk/blob/master/Examples/KlaviyoSwiftExamples/Shared/MenuPageViewController.swift#L35) |

Registering from `applicationDidBecomeActive(_:)` is advisable as it increases the chance of your user seeing the form. However, be advised that this will be shown as soon as the form is ready in the SDK, so you may still need to condition this based on the user's context within your application. Future versions of this product will provide more control in this regard.

**Note:** At this time, when device orientation changes any currently visible form is closed and will not be re-displayed automatically.

### Deep linking

> Deep linking to a particular screen based on user action from an IAF is similar to handling deep links originating from push notifications. [Step 3](#step-3-implement-handling-deep-links-in-your-app) of the deep linking section outlines exactly how this can be achieved. For further information on how the deep link is handled, see [apple's documentation](https://developer.apple.com/documentation/uikit/uiapplication/open(_:options:completionhandler:)).

Reviewer 4: Capitalize the A in Apple please!

Reviewer: ```suggestion
Deep linking to a particular screen based on user action from an IAF is similar to handling deep links originating from push notifications. [Step 3](#step-3-implement-handling-deep-links-in-your-app) of the deep linking section outlines exactly how this can be achieved. For further information on how the deep link is handled, see [Apple's documentation](https://developer.apple.com/documentation/uikit/uiapplication/open(_:options:completionhandler:)).
```
